### DISCUSSION TITLE
一个库里的数据导出来，想导到另一个库里，实现起来太麻烦

### DISCUSSION THREAD

**Discussion by chaixingyao-code** [taosdata/TDengine#4714] [General]
把test  Datebases里的所有数据导出来，
导入到另一个test2  Datebases里面
实现起来太麻烦，因为.sql文件里面使用的都是test，还的手动把test改成test2，要么就写脚本：

数据量大的话，根本实现不了。没有其他数据库那么方便，后期能不能加入这样的功能。让库之间的导入更方便些。

**Comment by sangshuduo** [comment]
请列举其他数据库提供的方法，我们评估一下

**Reply by chaixingyao-code** [reply]
linux下

一、导出数据库用mysqldump命令（注意mysql的安装路径，即此命令的路径）：
1、导出数据和表结构：
mysqldump -u用户名 -p密码 数据库名 > 数据库名.sql
#/usr/local/mysql/bin/   mysqldump -uroot -p abc > abc.sql
敲回车后会提示输入密码

2、只导出表结构
mysqldump -u用户名 -p密码 -d 数据库名 > 数据库名.sql
#/usr/local/mysql/bin/   mysqldump -uroot -p -d abc > abc.sql

注：/usr/local/mysql/bin/  --->  mysql的data目录


二、导入数据库
1、首先建空数据库
mysql>create database abc;

2、导入数据库
方法一：
（1）选择数据库
mysql>use abc;
（2）设置数据库编码
mysql>set names utf8;
（3）导入数据（注意sql文件的路径）
mysql>source /home/abc/abc.sql;
方法二：
mysql -u用户名 -p密码 数据库名 < 数据库名.sql
#mysql -uabc_f -p abc < abc.sql

**Comment by mujin-zl** [comment]
> 把test Datebases里的所有数据
> 导入来，导入到另一个test2 Datebases里面
> 实现起来太麻烦，因为.sql文件里面使用的都是test，还的手动把测试改成test2，只需写脚本：
> 
> 数据量大的话，根本实现不了。没有其他数据库那么方便，后续能不能加入这样的功能。让库之间的引入更方便些。

请问有什么好的解决方案了么
